catalog/transformer/catalog-transformer-propertyjson-metacard/src/main/java/ddf/catalog/transformer/metacard/propertyjson/PropertyJsonMetacardTransformer.java

```diff
@@ -0,0 +1,214 @@
+ * @author Ashraf Barakat
+ * @author ddf.isgs@lmco.com
+ * @see MetacardTransformer
+ * @see Metacard
+ * @see Attribute
+ */
+
+/**
+ * Transforms a metacard into a JSON with a format similiar to geojson, but without any
+ * special geojson components. Simply put, a flat mapping of all attributes to K/V pairs
+ * stored inside a root key of "property"
+ */
+public class PropertyJsonMetacardTransformer implements MetacardTransformer {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(PropertyJsonMetacardTransformer.class);
+
+    private static final String SOURCE_ID_PROPERTY = "source-id";
+
+    private static final String ISO_8601_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat(ISO_8601_DATE_FORMAT);
+
+    private final JsonSerializer json = JsonFactory.create(new JsonParserFactory(),
+            new JsonSerializerFactory().includeNulls()
+                    .includeEmpty()
+                    .includeBlank()
+                    .setJsonFormatForDates(false))
+            .serializer();
+
+    protected static final String METACARD_TYPE_PROPERTY_KEY = "metacard-type";
+
+    protected static final MimeType DEFAULT_MIME_TYPE = new MimeType();
+
+    public static final String ID = "propertyjson";
+
+    static {
+        try {
+            DEFAULT_MIME_TYPE.setPrimaryType("application");
+            DEFAULT_MIME_TYPE.setSubType("json");
+        } catch (MimeTypeParseException e) {
+            LOGGER.info("Failure creating MIME type", e);
+            throw new ExceptionInInitializerError(e);
+        }
+        DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("GMT"));
+    }
+
+    public static Map<String, Object> convertToJSON(Metacard metacard) throws CatalogTransformerException{
+        return convertToJSON(metacard, Collections.emptyList());
+    }
+
+    public static Map<String, Object> convertToJSON(Metacard metacard, List<AttributeType.AttributeFormat> dontInclude)
+            throws CatalogTransformerException {
+        if (metacard == null) {
+            throw new CatalogTransformerException("Cannot transform null metacard.");
+        }
+
+        Map<String, Object> rootObject = new HashMap<>();
+        Map<String, Object> properties = new HashMap<>();
+
+        for (AttributeDescriptor ad : metacard.getMetacardType()
+                .getAttributeDescriptors()) {
+
+            Attribute attribute = metacard.getAttribute(ad.getName());
+            if (attribute != null) {
+                Object value = convertAttribute(attribute, ad, dontInclude);
+                if (value != null) {
+                    properties.put(attribute.getName(), value);
+                }
+            }
+        }
+
+        properties.put(METACARD_TYPE_PROPERTY_KEY,
+                metacard.getMetacardType()
+                        .getName());
+
+        if (metacard.getSourceId() != null && !"".equals(metacard.getSourceId())) {
```

Comment:
You can use `StringUtils.isNotBlank(metacard.getSourceId())` here to simplify the logic